aptos/typescript/src/gas_station.ts

```diff
@@ -0,0 +1,188 @@
+        options: {
+            expireTimestamp: expirationSeconds
+        }
+    });
+    console.log("\nResponse from aptos.transaction.build.simple()");
+    console.log(transaction);
+    return transaction;
+}
+
+
+
+// Build, sponsor, sign, and execute a multiAgent Move script transaction
+async function buildSponsorSignAndSubmitMultiAgentTransaction(): Promise<CommittedTransactionResponse> {
+
+    // 1. Generate two funded accounts to act as sender and secondary signer
+    //    Replace function call with private key value printed to console after first run
+    const PRIVATE_KEY_ONE = await generateSingleKeyAccountPrivateKey(true); 
+    console.log("\nPRIVATE_KEY_ONE value:", PRIVATE_KEY_ONE);
+
+    const sender: SingleKeyAccount = new SingleKeyAccount({ privateKey: new Ed25519PrivateKey(PRIVATE_KEY_ONE) }); 
+    console.log("Multiagent transaction sender address:", sender.accountAddress.toString());
+
+    // replace function call with private key value printed to console after first run
+    const PRIVATE_KEY_TWO = await generateSingleKeyAccountPrivateKey(true); 
+    console.log("\nPRIVATE_KEY_TWO value:", PRIVATE_KEY_TWO);
+
+    const secondarySigner: SingleKeyAccount = new SingleKeyAccount({ privateKey: new Ed25519PrivateKey(PRIVATE_KEY_TWO) }); 
+    console.log("Multiagent transaction secondary signer address:", secondarySigner.accountAddress.toString());
+
+    // 2. Build a multiAgent transaction
+    let transaction = await buildMultiAgentScriptTransaction(sender.accountAddress, [secondarySigner.accountAddress]);
+
+    // 3. Sponsor the transaction with Shinami Gas Station
+    let feePayerAuthenticator = await gasStationClient.sponsorTransaction(transaction);
+    // Note that the SDK upates the transaction's feePayer address on a successful sponsorship
+    console.log("\ntransaction.feePayerAddress post-sponsorship:", transaction.feePayerAddress);
+
+    // 4. Generate the sender and secondary signer signatures
+    const senderAuthenticator = aptos.transaction.sign({ 
+        signer: sender, 
+        transaction 
+    });
+
+    const secondarySignerOneAuthenticator = aptos.transaction.sign({
+       signer: secondarySigner,
+       transaction 
+    }); 
+
+    // 5. Submit the transaction with the sender, seconardy signer, and feePayer signatures
+    const committedTransaction = await aptos.transaction.submit.multiAgent({
+        transaction,
+        senderAuthenticator,
+        additionalSignersAuthenticators: [secondarySignerOneAuthenticator],
+        feePayerAuthenticator: feePayerAuthenticator
+    });
+
+    // 6. Wait for and retrieve the executed transaction
+    const executedTransaction = await aptos.waitForTransaction({
+        transactionHash: committedTransaction.hash
+    });
+
+    return executedTransaction;
+
+}
+
+
+
+// Build a multi-agent script transaction with a fee payer.
+async function buildMultiAgentScriptTransaction(sender: AccountAddress, secondarySigners: Array<AccountAddress>, 
```

Comment:
Since `swap_coin.mv` only takes two signers, this function signature should match as opposed to taking an array.

Reply:
updated